src/core/Akka/Util/AkkaVersion.cs

```diff
@@ -0,0 +1,251 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="Vector.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2020 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2020 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Newtonsoft.Json;
+
+namespace Akka.Util
+{
+    /// <summary>
+    ///  Comparable version information.
+    ///
+    ///  The typical convention is to use 3 digit version numbers `major.minor.patch`,
+    ///  but 1 or two digits are also supported.
+    ///
+    ///  If no `.` is used it is interpreted as a single digit version number or as
+    ///  plain alphanumeric if it couldn't be parsed as a number.
+    ///
+    ///  It may also have a qualifier at the end for 2 or 3 digit version numbers such as "1.2-RC1".
+    ///  For 1 digit with qualifier, 1-RC1, it is interpreted as plain alphanumeric.
+    ///
+    ///  It has support for https://github.com/dwijnand/sbt-dynver format with `+` or
+    ///  `-` separator. The number of commits from the tag is handled as a numeric part.
+    ///  For example `1.0.0+3-73475dce26` is less than `1.0.10+10-ed316bd024` (3 < 10).
+    /// </summary>
+    public class AkkaVersion : IComparable<AkkaVersion>, IEquatable<AkkaVersion>
```

Comment:
Not sure about the name. Should be called just `Version`, but that would conflict everywhere with `System.Version`

Reply:
I can rename it `AppVersion` after merge - that way it wouldn't be confused with the Akka.NET version, which is what I initially thought when I saw the name.